CI failure looks real

<details>

```
FAILED: scipy/special/_cdflib.cp311-win_amd64.pyd.p/meson-generated__cdflib.c.obj 
"cc" "-Iscipy\special\_cdflib.cp311-win_amd64.pyd.p" "-Iscipy\special" "-I..\scipy\special" "-I..\..\..\..\..\..\..\..\hostedtoolcache\windows\Python\3.11.7\x64\Lib\site-packages\numpy\_core\include" "-IC:\hostedtoolcache\windows\Python\3.11.7\x64\Include" "-fvisibility=hidden" "-fdiagnostics-color=always" "-DNDEBUG" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-std=c99" "-O3" "-Wno-unused-but-set-variable" "-Wno-unused-function" "-Wno-conversion" "-Wno-misleading-indentation" "-mlong-double-64" "-D__USE_MINGW_ANSI_STDIO=1" "-DMS_WIN64=" "-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION" -MD -MQ scipy/special/_cdflib.cp311-win_amd64.pyd.p/meson-generated__cdflib.c.obj -MF "scipy\special\_cdflib.cp311-win_amd64.pyd.p\meson-generated__cdflib.c.obj.d" -o scipy/special/_cdflib.cp311-win_amd64.pyd.p/meson-generated__cdflib.c.obj "-c" scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_alngam':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:5050:37: error: 'M_PI' undeclared (first use in this function)
 5050 |   __pyx_v_offset = (0.5 * log((2. * M_PI)));
      |                                     ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:5050:37: note: each undeclared identifier is reported only once for each function it appears in
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_basym':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:5635:27: error: 'M_PI' undeclared (first use in this function)
 5635 |   __pyx_v_e0 = (2. / sqrt(M_PI));
      |                           ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_brcmp1':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:10753:36: error: 'M_PI' undeclared (first use in this function)
10753 |   __pyx_v_const = (1. / sqrt((2. * M_PI)));
      |                                    ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_brcomp':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:11602:37: error: 'M_PI' undeclared (first use in this function)
11602 |   __pyx_v_const = (1. / sqrt((2.0 * M_PI)));
      |                                     ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_cumnor':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:33774:41: error: 'M_PI' undeclared (first use in this function)
33774 |     __pyx_v_result = ((sqrt((1. / (2. * M_PI))) - __pyx_v_result) / __pyx_v_y);
      |                                         ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_dinvnr':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:35226:35: error: 'M_PI' undeclared (first use in this function)
35226 |   __pyx_v_r2pi = sqrt((1. / (2. * M_PI)));
      |                                   ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_gamma':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:43151:33: error: 'M_PI' undeclared (first use in this function)
43151 |   __pyx_v_d = (0.5 * (log((2. * M_PI)) - 1.0));
      |                                 ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_gratio':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:44808:36: error: 'M_PI' undeclared (first use in this function)
44808 |   __pyx_v_rt2pi = sqrt((1. / (2. * M_PI)));
      |                                    ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_psi':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:49194:31: error: 'M_PI' undeclared (first use in this function)
49194 |       __pyx_v_sgn = (((double)M_PI) / 4.0);
      |                               ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__pyx_f_5scipy_7special_7_cdflib_rcomp':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:49671:35: error: 'M_PI' undeclared (first use in this function)
49671 |   __pyx_v_r2pi = sqrt((1. / (2. * M_PI)));
      |                                   ^~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c: In function '__Pyx_ImportType_3_0_8':
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:51989:13: warning: unknown conversion type character 'z' in format [-Wformat=]
51989 |             "%s.%s size changed, may indicate binary incompatibility. "
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:51990:24: note: format string is defined here
51990 |             "Expected %zd from C header, got %zd from PyObject",
      |                        ^
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:51989:13: warning: unknown conversion type character 'z' in format [-Wformat=]
51989 |             "%s.%s size changed, may indicate binary incompatibility. "
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:51990:47: note: format string is defined here
51990 |             "Expected %zd from C header, got %zd from PyObject",
      |                                               ^
scipy/special/_cdflib.cp311-win_amd64.pyd.p/_cdflib.c:51989:13: warning: too many arguments for format [-Wformat-extra-args]
51989 |             "%s.%s size changed, may indicate binary incompatibility. "
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

</details>

--- a/scipy/special/_cdflib.pyx
+++ b/scipy/special/_cdflib.pyx
@@ -7,8 +7,7 @@
 import numpy as np
 cimport numpy as cnp
 cnp.import_array()
-from libc.math cimport sin, tan, log, exp, sqrt, floor
-from numpy.math cimport INFINITY, PI
+from libc.math cimport sin, tan, log, exp, sqrt, floor, INFINITY, pi as PI
 
 cdef double[3] spmpar = [np.finfo(np.float64).eps,
                          np.finfo(np.float64).tiny,

--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -168,11 +168,11 @@ py3.extension_module('_specfun',
   subdir: 'scipy/special'
 )
 
-cdflib_c = cython_gen.process('_cdflib.pyx')
-
 py3.extension_module('_cdflib',
-  cdflib_c,
+  cython_gen.process('_cdflib.pyx'),
+  c_args: cython_c_args,
   dependencies: [np_dep],
+  link_args: version_link_args,
   install: true,
   subdir: 'scipy/special'
 )

